### ISSUE TITLE
Unable to find devices on SoapyRemote on WSL2

### ISSUE THREAD

**Issue by FenixC4** [pothosware/SoapySDR#345]
I have configured my Windows 11 system to have a SoapySDRServer setup on Windows system and SoapySDR with SoapyRemoteModule built on WSL2. When I start the Server I am able to connect to it from Windows system itself and other deviecs in the network, however I am unable to connect to it from WSL2

![image](https://user-images.githubusercontent.com/43726835/156924333-9bcbb72e-9f58-4e47-9217-a52b3be209f3.png)

![image](https://user-images.githubusercontent.com/43726835/156924377-5f3e2eef-9641-448f-835f-f6470dc94410.png)

![image](https://user-images.githubusercontent.com/43726835/156924346-e7aed538-0322-41ef-baa9-e40526797f36.png)

What should I do to make it work ?

**Comment by zuckschwerdt** [comment]
Does `SoapySDRUtil --info` on WSL show the driver?

**Comment by FenixC4** [comment]
I'm using `./SoapySDRUtil` . because I have it installed in custom directory, I have done the same on my Ubuntu Laptop and it works just fine.

![image](https://user-images.githubusercontent.com/43726835/156931468-79bdd678-c78a-4a1a-94a0-3b6844578c20.png)

And as you can see it shows the driver

**Comment by zuckschwerdt** [comment]
Would have been too easy, right ;)
Next guess: with WSL2 the subsystem is a VM, the network interface is mapped to the host (Windows), but since it's external to the host it might need to pass through the firewall?
If you want to cut the remote client out, try `socat`. On WSL a `socat - tcp:192.168.1.109:1234` should connect and the SoapySDRServer should show that connection.

**Comment by FenixC4** [comment]
Ok, so I went down the rabbit hole and resolved the issue. It was all down to Windows's firewall.
First I found out that by default Windows doesn't allow WSL to connect to it in any way, so that event pings won't reach Windows system. 

By using this command `New-NetFirewallRule -DisplayName "WSL" -Direction Inbound  -InterfaceAlias "vEthernet (WSL)"  -Action Allow` you allow connections to be made from WSL to Windows machine. Since it only uses the HyperV adapter it is safe and closed from outside connections.

Next I had to add rules both Inbound and Outbound for the SoapyRemote itself since by default it got somehow blocked. After adding these rules everything works just fine.
